var/spack/repos/builtin/packages/doxygen/package.py

```diff
@@ -33,6 +33,30 @@ class Doxygen(CMakePackage):
     variant('mscgen', default=False,
             description='Build with support for code graphs from mscgen.')
 
+    executables = ['doxygen']
+
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('-v', output=str, error=str)
+        return output
```

Comment:
I think we should try to match a regex, similarly to what cmake does. This would hopefully increase robustness of detection.

Reply:
I'm not sure what that regex would be.  I checked CentOS, Fedora, Ubuntu, Alpine, and Gentoo and they all use `doxygen` without any prefixes or suffixes.  I'm not even sure the build file allows renaming the binary.

Reply:
Hmm, I think there is a misunderstanding. I meant we should use a regex on the output to extract the version - not on the name of the executable :slightly_smiling_face: Right now any garbage output that may be returned for any reson would be returned as a version.

Reply:
I think I fixed this now @alalazo